### PR TITLE
Consolidate android-impl into android-core

### PR DESCRIPTION
**Pull request by jakobuid**

### CODE REVIEW

**Review by Elyniss — APPROVED** [review]

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/pairing/engine/domain/PairingEngine.kt [hunk 1]
```diff
@@ -243,9 +200,7 @@ internal class PairingEngine(
             .onEach {
                 coroutineScope {
                     launch(Dispatchers.IO) {
-
                         println("kobe; Reconnect pairings")
```
Comment (Elyniss) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/pairing/engine/domain/PairingEngine.kt]: Pls remove

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/relay/RelayClient.kt [hunk 2]
```diff
@@ -20,27 +16,19 @@ import com.walletconnect.foundation.network.BaseRelayClient
...
-import timber.log.Timber
 
 object RelayClient : BaseRelayClient(), RelayConnectionInterface {
     private val connectionController: ConnectionController by lazy { wcKoinApp.koin.get(named(AndroidCommonDITags.CONNECTION_CONTROLLER)) }
     private val networkState: ConnectivityState by lazy { wcKoinApp.koin.get(named(AndroidCommonDITags.CONNECTIVITY_STATE)) }
     private val isNetworkAvailable: StateFlow<Boolean> by lazy { networkState.isAvailable }
     private val isWSSConnectionOpened: MutableStateFlow<Boolean> = MutableStateFlow(false)
 
-    fun initialize(relayServerUrl: String, connectionType: ConnectionType, application: Application) {
+
```
Comment (jakobuid) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/relay/RelayClient.kt:27]: add jvmstatic

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/CoreClient.kt [hunk 3]
```diff
@@ -25,20 +34,31 @@ object CoreClient {
         application: Application,
         relay: RelayConnectionInterface? = null
     ) {
-        if (relay != null) {
-            Relay = relay
+        plantTimber()
+
+        wcKoinApp.run {
```
Comment (jakobuid) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/CoreClient.kt:39]: Combine koin module and move RelayClient init above

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/pairing/client/PairingProtocol.kt [hunk 4]
```diff
@@ -17,15 +15,13 @@ import kotlinx.coroutines.delay
...
-import org.koin.dsl.module
 
 internal object PairingProtocol : PairingInterface {
     private lateinit var pairingEngine: PairingEngine
     private val logger: Logger by lazy { wcKoinApp.koin.get() }
 
-    internal fun initialize(metaData: Core.Model.AppMetaData) {
+    internal fun initialize() {
         pairingEngine = wcKoinApp.koin.get()
-        wcKoinApp.modules(module { with(metaData) { single { AppMetaData(name, description, url, icons, Redirect(redirect)) } } })
```
Comment (jakobuid) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/pairing/client/PairingProtocol.kt:28]: Add pairing.setup()

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/CommonModule.kt [hunk 5]
```diff
@@ -16,7 +16,7 @@ import timber.log.Timber
 import kotlin.reflect.jvm.jvmName
 import com.walletconnect.foundation.di.commonModule as foundationCommonModule
 
-fun commonModule() = module {
+fun coreCommonModule() = module {
```
Comment (TalhaAli00) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/CommonModule.kt:19]: Change filename to CoreCommonModule

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/CryptoModule.kt [hunk 6]
```diff
@@ -16,7 +20,7 @@ import org.koin.core.scope.Scope
 import org.koin.dsl.module
 import java.io.File
 
-fun androidApiCryptoModule() = module {
+fun coreCryptoModule() = module {
```
Comment (TalhaAli00) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/CryptoModule.kt:23]: Change filename to CoreCryptoModule

Comment (TalhaAli00) [reply androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/CryptoModule.kt:23]: Add JvmSynthetic

File: .github/workflows/ci_sdks.yml [hunk 7]
```diff
@@ -25,7 +25,7 @@ jobs:
       matrix:
         conf: [
           { name: Android-Core SDK, command: ":androidCore:sdk:testDebugUnitTest" },
-          { name: Android-Core Impl, command: ":androidCore:impl:testDebugUnitTest" },
+#          { name: Android-Core Impl, command: ":androidCore:impl:testDebugUnitTest" },
```
[the remark this answers is not in the document]
Comment (jakobuid) [reply .github/workflows/ci_sdks.yml:28]: addressed

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/CommonModule.kt (hunk 5 above)
Comment (jakobuid) [reply androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/CommonModule.kt:19]: addressed

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/CryptoModule.kt (hunk 6 above)
Comment (jakobuid) [reply androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/CryptoModule.kt:23]: addressed

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/CoreClient.kt (hunk 3 above)
Comment (jakobuid) [reply androidCore/sdk/src/main/kotlin/com/walletconnect/android/CoreClient.kt:39]: addressed

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/CoreClient.kt [hunk 8]
```diff
@@ -25,20 +34,23 @@ object CoreClient {
...
-            module {
-                single { PairingEngine() }
-                single { Pairing }
-                single<PairingControllerInterface> { PairingController }
-                single { Relay }
-            }
-        )
-        PairingProtocol.initialize(metaData)
+
+        RelayClient.initialize(relayServerUrl, connectionType)
```
Comment (TalhaAli00) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/CoreClient.kt:52]: Don't we still want to only initialize if we are using our RelayClient. If developers pass their own ,then we don't want to do this right?

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/model/type/Sequence.kt [hunk 9]
```diff
@@ -0,0 +1,9 @@
+package com.walletconnect.android.internal.common.model.type
+
+import com.walletconnect.android.internal.common.model.Expiry
+import com.walletconnect.foundation.common.model.Topic
+
+interface Sequence {
```
Comment (TalhaAli00) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/model/type/Sequence.kt:6]: Is this used anywhere outside of Sign? Maybe we can move it?

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/utils/Timber.kt [hunk 10]
```diff
@@ -0,0 +1,18 @@
+package com.walletconnect.android.utils
+
+import com.walletconnect.android.BuildConfig
+import timber.log.Timber
+
+internal fun plantTimber() {
```
Comment (TalhaAli00) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/utils/Timber.kt:6]: Should we move this into where we setup the logger in koin?

File: auth/sdk/src/main/kotlin/com/walletconnect/auth/di/StorageModule.kt [hunk 11]
```diff
@@ -1,10 +1,9 @@
 package com.walletconnect.auth.di
 
 
-import com.walletconnect.android.impl.di.coreStorageModule
 import org.koin.dsl.module
 
 @JvmSynthetic
 internal fun storageModule() = module {
```
Comment (TalhaAli00) [inline auth/sdk/src/main/kotlin/com/walletconnect/auth/di/StorageModule.kt:7]: I think we can remove this file

File: build.gradle.kts [hunk 12]
```diff
@@ -59,7 +59,7 @@ task<Delete>("clean") {
 
 nexusPublishing {
     repositories {
-//        project.version = "-SNAPSHOT"
+        project.version = "-SNAPSHOT"
```
Comment (TalhaAli00) [inline build.gradle.kts:62]: let's comment this back out

File: buildSrc/src/main/kotlin/Versions.kt [hunk 13]
```diff
@@ -6,10 +6,10 @@ const val KEY_SDK_NAME = "SDK_NAME"
 const val dokkaVersion = "1.6.21"
 
 //Latest versions
-const val BOM_VERSION = "1.1.0"
-const val FOUNDATION_VERSION = "1.2.0"
-const val CORE_VERSION = "1.6.0"
-const val SIGN_VERSION = "2.4.0"
+const val BOM_VERSION = "1.2.0-SNAPSHOT"
```
Comment (TalhaAli00) [inline buildSrc/src/main/kotlin/Versions.kt:9]: remove snapshot from versions

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/client/ChatProtocol.kt [hunk 14]
```diff
@@ -34,7 +33,7 @@ internal class ChatProtocol : ChatInterface {
             wcKoinApp.run {
                 modules(
                     commonModule(),
-                    cryptoModule(),
+//                    cryptoModule(),
```
Comment (TalhaAli00) [inline chat/sdk/src/main/kotlin/com/walletconnect/chat/client/ChatProtocol.kt:36]: remove

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/di/CommonModule.kt [hunk 15]
```diff
@@ -10,14 +10,13 @@ import com.walletconnect.chat.common.adapter.JsonRpcResultAdapter
 import org.koin.core.qualifier.named
 import org.koin.dsl.module
 import kotlin.reflect.jvm.jvmName
-import com.walletconnect.android.internal.common.di.commonModule as androidCommonModule
 
 @JvmSynthetic
 internal fun commonModule() = module {
 
-    includes(androidCommonModule())
+//    includes(androidCommonModule())
```
Comment (TalhaAli00) [inline chat/sdk/src/main/kotlin/com/walletconnect/chat/di/CommonModule.kt:17]: remove

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/di/JsonRpcModule.kt [hunk 16]
```diff
@@ -7,12 +7,11 @@ import com.walletconnect.chat.json_rpc.JsonRpcMethod
 import com.walletconnect.utils.addDeserializerEntry
 import com.walletconnect.utils.addSerializerEntry
 import org.koin.dsl.module
-import com.walletconnect.android.impl.di.jsonRpcModule as coreJsonRpcModule
 
 @JvmSynthetic
 internal fun jsonRpcModule() = module {
 
-    includes(coreJsonRpcModule())
+//    includes(coreJsonRpcModule())
```
Comment (TalhaAli00) [inline chat/sdk/src/main/kotlin/com/walletconnect/chat/di/JsonRpcModule.kt:14]: remove

File: foundation/src/main/kotlin/com/walletconnect/foundation/di/CommonModule.kt [hunk 17]
```diff
@@ -14,7 +14,7 @@ import org.koin.core.qualifier.named
 import org.koin.dsl.module
 import kotlin.reflect.jvm.jvmName
 
-fun commonModule() = module {
+fun foundationCommonModule() = module {
```
Comment (TalhaAli00) [inline foundation/src/main/kotlin/com/walletconnect/foundation/di/CommonModule.kt:17]: rename file

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/CoreClient.kt (hunk 8 above)
Comment (jakobuid) [reply androidCore/sdk/src/main/kotlin/com/walletconnect/android/CoreClient.kt]: Ofc, good catch

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/utils/Timber.kt (hunk 10 above)
Comment (jakobuid) [reply androidCore/sdk/src/main/kotlin/com/walletconnect/android/utils/Timber.kt:6]: I'm not a fun of moving it somewhere else

File: auth/sdk/src/main/kotlin/com/walletconnect/auth/di/StorageModule.kt (hunk 11 above)
Comment (jakobuid) [reply auth/sdk/src/main/kotlin/com/walletconnect/auth/di/StorageModule.kt:7]: Indeed

File: buildSrc/src/main/kotlin/Versions.kt (hunk 13 above)
Comment (jakobuid) [reply buildSrc/src/main/kotlin/Versions.kt]: Versions are up to date after merging in develop

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/model/type/Sequence.kt (hunk 9 above)
Comment (jakobuid) [reply androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/model/type/Sequence.kt]: Yes, It's used in Pairing

**Review by Elyniss — APPROVED** [review]

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/CoreClient.kt [hunk 18]
```diff
@@ -1,20 +1,29 @@
...
 import com.walletconnect.android.relay.RelayConnectionInterface
+import com.walletconnect.android.utils.plantTimber
+import com.walletconnect.android.utils.projectId
+import org.koin.android.ext.koin.androidContext
 import org.koin.dsl.module
 
 object CoreClient {
     val Pairing: PairingInterface = PairingProtocol
-    var Relay: RelayConnectionInterface = RelayClient
+    val Relay: RelayConnectionInterface = RelayClient
```
Comment (TalhaAli00) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/CoreClient.kt:26]: If this is a val, how would a developer get access to their own Relay implementation?

**Review by TalhaAli00 — APPROVED** [review]

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/CoreClient.kt (hunk 18 above)
Comment (jakobuid) [reply androidCore/sdk/src/main/kotlin/com/walletconnect/android/CoreClient.kt:26]: They pass it with the initialize function, and we grab it and inject it into our koin tree. It's an external component so the can call it themselves

**Pull request merged**